File: backend/ReptiRealm/src/ReptiRealm/Models/Feed.cs
using System.ComponentModel.DataAnnotations;

namespace ReptiRealm.Models
{
    public class Feed : BaseEntity
    {
        [Required]
        public required DateOnly Date { get; set; }
        [Required]
        public required int Number { get; set; }
        [Required]
        public required bool Eaten { get; set; }
        public string? Comment { get; set; }
        [Required]
        public required virtual FoodType FoodType { get; set; }
    }
}

File: App/lib/models/feed.dart
import 'package:reptirealm/models/food_types.dart';
import 'package:reptirealm/models/regurgitation.dart';

class Feed {
  final String? id;
  final DateTime date;
  final int number;
  final bool eaten;
  final String? comment;
  final FoodType foodType;
  final Regurgitation? regurgitation;

  Feed({
    this.id,
    required this.date,
    required this.number,
    required this.eaten,
    this.comment,
    required this.foodType,
    this.regurgitation,
  });
}